src/lib/services/setting-service.ts

```diff
@@ -44,20 +44,27 @@ export default class SettingService {
 
     async insert(id: string, value: object, createdBy: string): Promise<void> {
         const exists = await this.settingStore.exists(id);
+        let data = value;
+        if (typeof value === 'object') {
+            data = { id, ...value };
+        } else {
+            this.logger.error('Invalid setting value, source: %s', id);
+        }
```

Comment:
I think this can cause subtle bugs. If the incoming data is a string for example we won't add the data in the event. I think I would either:

(A) Check all incoming paths to this code to verify that we are indeed validating that it is an object. If it is, then what we have done here is fine.
(B) If we are not sure whether or not it's an object I'd prefer something like: 

```
 if (typeof value === 'object') {
            data = { id, ...value };
        } else {
            data = { id, value } // this was the original right?
        }
```

Reply:
The original was just id

Reply:
but you are right, the code is wrong

Reply:
patched it now